app/src/main/java/com/babylon/wallet/android/presentation/transaction/TransactionReviewViewModel.kt

```diff
@@ -484,24 +491,23 @@ sealed interface PreviewType {
             override val from: List<AccountWithTransferableResources>,
             override val to: List<AccountWithTransferableResources>,
             val pools: List<com.babylon.wallet.android.domain.model.resources.Pool>,
+            val poolsWithAssociatedDapps: Map<String, DApp>,
```

Comment:
What if we had `val poolsWithAssociatedDapps: Map<Pool, DApp?>` instead of having this + pools?

Reply:
we need to have all the pools to count how many of them is unknown, since pools associated with dapps are only the ones which have the association. I could get rid of that List<Pool>  having  Map<Pool, Dapp?>.